app/collections/collaborators.jsx

```diff
@@ -285,50 +286,80 @@ export class CollectionCollaborators extends React.Component {
     .then((hasSettingsRole) => {
       this.setState({ hasSettingsRole });
     });
-    this.update();
+    this.update(this.props.location.query.page || 1);
   }
 
-  update() {
-    apiClient.type('collection_roles').get({ collection_id: this.props.collection.id })
-      .then((roleSets) => {
-        this.setState({ roleSets });
-      });
+  componentWillReceiveProps(nextProps) {
+    if (this.props.location.query.page !== nextProps.location.query.page) {
+      this.update(nextProps.location.query.page);
+    }
+  }
+
+  update(_page = 1) {
+    if (_page >= 1) {
```

Comment:
Maybe `_page = parseInt(_page)` somewhere here to avoid string/number bugs.

Reply:
Unless this is one of those cases where the API client requires numbers as strings, in which case ignore me.